RecoLocalCalo/CaloTowersCreator/src/CaloTowersCreator.cc

```diff
@@ -62,7 +63,8 @@ CaloTowersCreator::CaloTowersCreator(const edm::ParameterSet& conf) :
         conf.getParameter<double>("MomHBDepth"),
         conf.getParameter<double>("MomHEDepth"),
         conf.getParameter<double>("MomEBDepth"),
-        conf.getParameter<double>("MomEEDepth")
+        conf.getParameter<double>("MomEEDepth"),
+        conf.exists("HcalPhase") ? conf.getParameter<int>("HcalPhase") : 0
```

Comment:
fillDescriptions is mainly to declare the parameters so that they are known to the framework; the fact that their values will be different in different configurations is natural.
The point is to have the HCalPhase declared so that its value makes to the provenance when it's not listed in some config.